# hunk 1: multi-inheritance is broken https://sourceforge.net/p/swig/bugs/1379/
# hunk 2: compilation issue on GCC 4.6 http://sourceforge.net/p/swig/patches/296/
# hunk 3: thread bug on iterators https://sourceforge.net/p/swig/bugs/1370/
--- eossdk_wrap.cpp-t
+++ eossdk_wrap.cpp-t
@@ -1599,6 +1599,10 @@
     PyErr_SetString(PyExc_TypeError, "Attempt to append a non SwigPyObject");
     return NULL;
   }
+  while (sobj->next && SwigPyObject_Check(sobj->next)) {
+     sobj = (SwigPyObject *)sobj->next;
+  }
+  ((SwigPyObject *)next)->next = sobj->next;
   sobj->next = next;
   Py_INCREF(next);
   return SWIG_Py_Void();
@@ -3403,6 +3407,7 @@
 
 
 #include <stdexcept>
+#include <cstddef>
 
 
 namespace swig {
@@ -3518,7 +3523,11 @@
     }
       
   public:
-    virtual ~SwigPyIterator() {}
+    virtual ~SwigPyIterator() {
+       SWIG_PYTHON_THREAD_BEGIN_BLOCK;
+       _seq = 0;
+       SWIG_PYTHON_THREAD_END_BLOCK;
+    }
 
     // Access iterator method, required by Python
     virtual PyObject *value() const = 0;
